runtime/src/in_mem_accounts_index.rs

```diff
@@ -330,78 +356,65 @@ impl<T: IndexValue> InMemAccountsIndex<T> {
         reclaims: &mut SlotList<T>,
         previous_slot_entry_was_cached: bool,
     ) {
-        // try to get it just from memory first using only a read lock
-        self.get_only_in_mem(pubkey, |entry| {
-            if let Some(entry) = entry {
+        let mut m = Measure::start("entry");
+        let mut map = self.map().write().unwrap();
```

Comment:
Note grabbing the write lock for the entire bucket is now required for individual account updates as well. Will now compete with any reads to the same bucket from RPC/clean/shrink.

Same concern as in https://github.com/solana-labs/solana/pull/21549/files#r766298457 for large buckets and expensive scans